src/transformState.js

```diff
@@ -1,11 +1,22 @@
 'use strict';
 
-/**
- * @param {Object} state
- * @param {Object[]} actions
- */
 function transformState(state, actions) {
-  // write code here
+  for (const key of actions) {
+    switch (key.type) {
+      case 'addProperties':
+        Object.assign(state, key.extraData);
+        break;
+
+      case 'removeProperties':
+        key.keysToRemove.forEach(el => delete state[el]);
+        break;
+
+      default:
```

Comment:
Consider changing this condition to "case" and use "default" to throw an error (checklist -> "should always have default case for error handling.")

Reply:
good point! It's already changed.